src/transformers/models/funnel/modeling_tf_funnel.py

```diff
@@ -652,10 +661,11 @@ def call(
         for block_index, block in enumerate(self.blocks):
             pooling_flag = shape_list(hidden)[1] > (2 if self.separate_cls else 1)
             pooling_flag = pooling_flag and block_index > 0
+            pooled_hidden = self.attention_structure.pool_tensor(hidden, mode=self.attention_structure.pooling_type)
```

Comment:
this seems to slightly change the logic here - is this OK?

Reply:
maybe a comment here would be great

Reply:
The problem was coming from `shift = 2 if q_head.shape[1] != context_len else 1`. Here, `shift` in graph mode becomes an undefined tensor because it can be either 1 or 2. Which makes the line `r = position_embeds[self.block_index][shift - 1]` impossible to be compiled because in a graph, TF cannot mix tensors and numbers, here (`shift -1`)

Reply:
If the method `self.attention_structure.pre_attention_pooling` is not changed as suggested above, this line should be removed (not sure how it's linked to the code you mention which is in `self.attention_structure.post_attention_pooling`).

Otherwise, the line should be inside the test `if pooling_flag` otherwise it breaks the current behavior.